tests/failpoints/cases/test_snap.rs

```diff
@@ -545,24 +545,37 @@ fn test_sending_fail_with_net_error() {
     let mut cluster = new_server_cluster(1, 2);
     configure_for_snapshot(&mut cluster);
 
-    let on_send_store_fp = "receiving_snapshot_net_error";
-
     let pd_client = Arc::clone(&cluster.pd_client);
     // Disable default max peer count check.
     pd_client.disable_default_operator();
     cluster.run_conf_change();
 
     cluster.must_put(b"k1", b"v1");
 
-    // store2 will not receive any data from stream
-    fail::cfg(on_send_store_fp, "return()").unwrap();
-    pd_client.add_peer(1, new_learner_peer(2, 2));
+    // should ensure leader can send the snapshot request
+    let (notify_tx, notify_rx) = mpsc::channel();
+    cluster.sim.write().unwrap().add_send_filter(
+        1,
+        Box::new(SendMsgOnceFilter::new(notify_tx, MessageType::MsgSnapshot)),
+    );
+    // should ensure learner can receive the snapshot request
+    let (receive_tx, receive_rx) = mpsc::channel();
+    cluster.sim.write().unwrap().add_recv_filter(
+        2,
+        Box::new(SendMsgOnceFilter::new(receive_tx, MessageType::MsgSnapshot)),
+    );
```

Comment:
If `receiving_snapshot_net_error` failpoint turns on, peer 2 should not receive `MsgSnapshot`.
Normally the code here should be blocked forever.`receive_rx.recv_timeout(Duration::from_secs(3)).unwrap()`
However, as the comment above said, there is a bug in `SendMsgOnceFilter` so the block does not happen.

Reply:
the test arim that peer2 will interrupt in receiving snapshot to test the receiving count is right.